python/EMR_SECURITY_GROUPS_RESTRICTED/EMR_SECURITY_GROUPS_RESTRICTED.py

```diff
@@ -116,29 +116,39 @@ def get_sg_cluster_mapping(cluster_list, emr_client):
                 sg_list_combined.add(security_group)
     return sg_cluster_map, sg_list_combined
 
-def get_open_security_groups(sg_list, ec2_client):
+def get_open_security_groups(sg_list, config_client):
     open_sg_list = set([])
-    paginator = ec2_client.get_paginator('describe_security_groups')
-    page_iterator = paginator.paginate(GroupIds=list(sg_list))
-    for page in page_iterator:
-        for security_group in page['SecurityGroups']:
-            if is_sg_open_ipv4(security_group):
-                open_sg_list.add(security_group['GroupId'])
-            elif is_sg_open_ipv6(security_group):
-                open_sg_list.add(security_group['GroupId'])
+    resource_keys = []
+    for security_group in sg_list:
+        resource_key = {
+            "resourceType": "AWS::EC2::SecurityGroup",
+            "resourceId": security_group
+        }
+        resource_keys.append(resource_key)
+    res = config_client.batch_get_resource_config(resourceKeys=resource_keys)
```

Comment:
Made changes as per comments 1 & 3
Regarding 2, consider the following:
i) If the resource has just been created, it won't be picked up by Config for some time. During this time, no configuration item will be returned for the resource.
ii) If the resource has been deleted recently, it will be a part of Config's resources for some time before it is marked as deleted. Post then, no config item will be returned.

In both the cases, no error will be thrown. Consequently, I feel we can do away with error handling and leave the change to be reflected in the next execution of the Config rule.

@jongogogo What are your views?